src/sql/base/browser/ui/infoBox/infoBox.ts

```diff
@@ -96,6 +127,51 @@ export class InfoBox extends Disposable implements IThemable {
 		}
 	}
 
+	public get isClickable(): boolean {
+		return this._isClickable;
+	}
+
+	public set isClickable(v: boolean) {
+		this._isClickable = v;
+		if (this._isClickable) {
+			this._clickableIndicator.style.display = '';
+			this._clickableIndicator.tabIndex = 0;
+			this._infoBoxElement.style.cursor = 'pointer';
+			this._infoBoxElement.setAttribute('role', 'button');
+			this._textElement.style.maxWidth = 'calc(100% - 75px)';
+			this.registerClickListeners();
+		} else {
+			this._clickableIndicator.style.display = 'none';
+			this._clickableIndicator.tabIndex = -1;
+			this._infoBoxElement.style.cursor = 'default';
+			this._infoBoxElement.removeAttribute('role');
+			this._textElement.style.maxWidth = '';
+			this.unregisterClickListeners();
+		}
+	}
+
+	private registerClickListeners() {
+		this._onDidClick = this._register(new Emitter<undefined>());
+		this._register(DOM.addDisposableListener(this._infoBoxElement, DOM.EventType.CLICK, e => {
+			if (this._isClickable) {
+				this._onDidClick.fire(undefined);
+			}
+		}));
+
+		this._register(DOM.addDisposableListener(this._infoBoxElement, DOM.EventType.KEY_PRESS, e => {
+			const event = new StandardKeyboardEvent(e);
+			if (this._isClickable && (event.equals(KeyCode.Enter) || !event.equals(KeyCode.Space))) {
+				this._onDidClick.fire(undefined);
+				DOM.EventHelper.stop(e);
+				return;
+			}
+		}));
+	}
+
+	private unregisterClickListeners() {
+		this.dispose();
```

Comment:
You shouldn't dispose the whole object - just setting clickable to false doesn't mean the whole thing is disposed. Just save the disposable listeners in a DisposableStore and call clear on it here.